Documentation/project-docs/developer-guide.md

```diff
@@ -38,21 +37,38 @@ the managed build which produces the MSIL code and nuget packages that make up C
 Calling the script `build` attempts to build both the native and managed code.
 Only use it when the parameters that you are passing to the script apply for both components. Otherwise, use the scripts `build-native` and `build-managed` respectively.
 
-**Examples**
+Most of the build configurations are defaulted based on where you are building but here are a few common parameters to customize the build configuration:
 
-- Building in debug mode for platform x64
+- `-framework` identifies the target framework for the build. It defaults to `netcoreapp` but possible values include `netcoreapp`, `netfx` or `uap`. (msbuild property `TargetGroup`)
+- `-os` identifies the OS for the build. It defaults to the OS you are running on but possible values include `Windows_NT`, `Unix`, `Linux`, or `OSX`. (msbuild property `OSGroup`)
+- `-debug|-release` controls the optimization levels the compilers use for the build. It defaults to `Debug`. (msbuild property `ConfigurationGroup`)
+- `-buildArch` identifies the architecture for the build. It defaults to `x64` but possible values include `x64`, `x86`, `arm`, or `arm64`. (msbuild property `ArchGroup`)
+
+These options are common for build, build-managed, build-native, and build-tests scripts.
+
+For more details on the build configurations see [project-guidelines](../coding-guidelines/project-guidelines.md).
+
+*Note*: Before working on individual projects or test projects you **must** run `build.cmd` once before beginning that work.
```

Comment:
Same note to add as above -- suggestion to run build.cmd if msbuild fails unexpectedly afer a pull.